### PR TITLE
Improve deployment UX

### PR DESCRIPTION
**Pull request by johananl**
- Restore manifests removed in https://github.com/instaclustr/cassandra-operator/pull/283.
- Add a Make target for generating deployment bundles for CRDs and the operator (`make bundle`).
- Update `op_guide.md`.
- Include `serviceAccountName` in examples.

### CODE REVIEW

File: doc/op_guide.md [hunk 1]
```diff
@@ -1,32 +1,17 @@
...
 
 ### Deploy the Operator
 
  1) Deploy the [CRDs][crds] used by the operator to manage Cassandra clusters:
 
     ```
-    kubectl apply -f deploy/crds/cassandraoperator_v1alpha1_cassandrabackup_crd.yaml
-    kubectl apply -f deploy/crds/cassandraoperator_v1alpha1_cassandracluster_crd.yaml
-    kubectl apply -f deploy/crds/cassandraoperator_v1alpha1_cassandradatacenter_crd.yaml
+    kubectl apply -f deploy/crds.yaml
```
[the remark this answers is not in the document]
Comment (johananl) [reply doc/op_guide.md]: I planned `make bundle` to be something the maintainers do, not the users. As you can see, the resulting bundle files are already committed. We should re-run `make bundle` when something changes in the individual manifests and then commit the result.

[the remark this answers is not in the document]
Comment (johananl) [reply doc/op_guide.md]: In general, my aim is to reduce the things the user has to worry about to the absolute minimum as this has a huge impact on UX and the users' ability to adopt the operator.

[the remark this answers is not in the document]
Comment (alourie) [reply doc/op_guide.md]: That's a fairly good point. I'll add a CI check that runs this and fails if not committed. Thanks a lot!

**Comment by smiklosovic** [comment]
https://github.com/instaclustr/cassandra-operator/commit/a2941ed6b9b6238d6974ab21c84a04844639911b

**Pull request closed without merging**